gpytorch/distributions/new_multitask_multivariate_normal.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+import math
+from collections import Counter
+from typing import Iterable, Optional, Tuple
+
+import torch
+from torch import Tensor
+from torch.distributions.multivariate_normal import _batch_mahalanobis
+
+from gpytorch.distributions import MultivariateNormal
+
+ADMISSIBLE_ORDERS = {"nnt", "ntt", "ntn", "ttn", "tnt", "tnn"}
+
+
+class MultitaskMultivariateNormal(MultivariateNormal):
+    def __init__(self, mean: Tensor, covariance: Tensor, order: str = "ntnt") -> MultitaskMultivariateNormal:
+        """Construct a MultitaskMultivariateNormal from a stacked representation of its covariance matrix.
+
+        :param torch.tensor mean: A `batch_shape x n x t` tensor of means
+        :param torch.tensor covariance: A `batch_shape x shape`-dim tensor, where `shape` has either three
+            or four elements and can be any combination of "n" and "t" with at least one and at most two
+            occurrences of both "n" and "t" (currently only n==2 is supported)
+
+        Internally, the covariance is represented consistently as `n x t n x t` or `t x n x n` (case of
```

Comment:
That makes sense. The reason for me using `n x t x n x t` initially was that it's straightforward to get to a full `nt x nt` covariance matrix with a single `view` / `reshape` without doing much permutation acrobatics: `self._covar.view(*self._batch_shape, n * t, n * t)`. But I guess that's not all that important. 

Re using a `t x n x n` `KroneckerProductLazyTensor` - In this case I wouldn't actually use a `KroneckerProductLazyTensor`, we can just compute the log-likelihood in a batched fashion for each `n x n` matrix and sum them up. Using the `KroneckerProductLazyTensor` would probably create more overhead than it would help. For `t x t x n x n` it makes sense, of course.

Reply:
Hmm, actually, thinking about this more, for Kronecker strucutre we can be a smarter rather than just expressing the `t x t x n x n` from the Kronecker representation (see e.g. the related https://github.com/cornellius-gp/gpytorch/pull/1086). So we'd actually want to just keep a `t x t` and a `n x n` tensor around, then we don't even need to reason about `t x t x n x n`.